frontend/rust-lib/flowy-core/src/deps_resolve/search_deps.rs

```diff
@@ -0,0 +1,18 @@
+use flowy_search::folder::handler::FolderSearchHandler;
+use flowy_search::folder::indexer::FolderIndexManager;
+use flowy_search::services::manager::SearchManager;
+use std::sync::{Arc, Weak};
+
+pub struct SearchDepsResolver();
+impl SearchDepsResolver {
+  pub async fn resolve(folder_indexer: Weak<FolderIndexManager>) -> Arc<SearchManager> {
```

Comment:
Can user Arc<FolderIndexManager> here. then you don't need to do upgrading. Weak is used when you are going to spawn a task or something like that